### ISSUE TITLE
Unable to load dynamic library 'intl.so'

### ISSUE THREAD

**Issue by ramsey** [shivammathur/setup-php#739]
**Describe the bug**

When I try installing the `intl` extension on PHP 8.2, I get the following error:

```
PHP Warning:  PHP Startup: Unable to load dynamic library 'intl.so' (tried: /usr/lib/php/20220829-zts/intl.so (/usr/lib/php/20220829-zts/intl.so: undefined symbol: executor_globals), /usr/lib/php/20220829-zts/intl.so.so (/usr/lib/php/20220829-zts/intl.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0
```

This issue sounds similar to #722, but I'm not sure if it's the same problem.

**Version**
- [x] I have checked releases, and the bug exists in the latest patch version of `v1` or `v2`.
- [x] `v2`
- [ ] `v1`

**Runners**

- [x] GitHub Hosted
- [ ] Self Hosted

**Operating systems**

- Ubuntu

**PHP versions**

- PHP 8.2

**To Reproduce**

Here's the step I'm using in my GitHub Actions yml file:

```yaml
      - name: "Install PHP"
        uses: "shivammathur/setup-php@2.25.2"
        with:
          php-version: "latest"
          extensions: "intl-73.1"
          coverage: "none"
        env:
          debug: true
          phpts: "ts"
```

**Expected behavior**

The `intl` extension should load.

**Screenshots/Logs**

**Additional context**

I've tried this with ICU version 67.1 and 73.1 (trying to see if maybe 67.1 had trouble on PHP 8.2).

You can see the failed runs for both, here:

- https://github.com/ramsey/php-ecma-intl-ext/actions/runs/5195176011/jobs/9367634410
- https://github.com/ramsey/php-ecma-intl-ext/actions/runs/5195212718/jobs/9367702771

**Are you willing to submit a PR?**

I'm willing to submit one if you give me a few pointers for where to get started. 🙂

**Comment by shivammathur** [comment]
Hi Ben, thanks for the sponsorship 🙂

I have added support in for zts intl builds in https://github.com/shivammathur/setup-php/commit/12282c9597ef0c7329c0f382c5953c6248a43a62 and https://github.com/shivammathur/icu-intl/commit/d6ee42cd39943b532979d8f8ad6223042347b5cc.

This will be in the next release.

**Comment by ramsey** [comment]
Thanks!

**Comment by shivammathur** [comment]
Released `2.25.3` with the fix.
https://github.com/shivammathur/setup-php/releases/tag/2.25.3
